websites/A/Anime4up/presence.ts

```diff
@@ -1,22 +1,144 @@
-const presence = new Presence({
-  clientId: "770030754356396052"
-});
-
-let video = {
-  duration: 0,
-  currentTime: 0,
-  paused: true
+type Functionlize<T> = {
+  [P in keyof T]: () => T[P];
 };
 
-function getTimestamps(
-  videoTime: number,
-  videoDuration: number
-): Array<number> {
-  const startTime = Date.now(),
-    endTime = Math.floor(startTime / 1000) - videoTime + videoDuration;
-  return [Math.floor(startTime / 1000), endTime];
+interface Route extends Functionlize<Partial<PresenceData>> {
+  path: RegExp;
+  playback?(): boolean;
+  run?(): PresenceData;
+}
+
+enum PresenceSettings {
+  TIMESTAMP = "timestamp",
+  BUTTONS = "buttons",
+  LOGO = "logo"
+}
+enum Icons {
+  PAUSED = "paused",
+  PLAYED = "played",
+  SEARCHING = "searching",
+  LOCATION = "location",
+  DISCOVERY = "discovery"
 }
 
+enum Logos {
+  LIGHT = "light-logo",
+  DARK = "dark-logo"
+}
+
+let video = { duration: 0, currentTime: 0, paused: true };
+
+const presence = new Presence({
+    clientId: "770030754356396052"
+  }),
+  getTimestamps = (videoTime: number, videoDuration: number): number[] => {
+    const startTime = Date.now(),
+      endTime = Math.floor(startTime / 1000) - videoTime + videoDuration;
+    return [Math.floor(startTime / 1000), endTime];
+  },
```

Comment:
This function already exists in `Presence` class

Reply:
nice. didn't know about it